packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3828,21 +3846,37 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
           canRemoveOrAdd = true;
           break;
         case _RouteLifecycle.pop:
+          entry.handlePop(
+            navigator: this,
+            previousPresent: _getRouteBefore(index, _RouteEntry.willBePresentPredicate)?.route,
+          );
+          if (entry.currentState == _RouteLifecycle.idle) {
```

Comment:
Maybe instead of inferring this from the state, make handlePop return true/false indicating whether the pop was successful?

Reply:
Sounds good